### PR TITLE
[IR] Fixed bug with SharedVariablesManager

### PR DESCRIPTION
**Pull request by homuroll**

### CODE REVIEW

File: backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt [hunk 1]
```diff
@@ -0,0 +1 @@
+fun main() = println(A().x)
```
Comment (SvyatoslavScherbina) [inline backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt:1]: Please prefer making tests self-containing. Using `assertEquals` should help.

Comment (homuroll) [reply backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt:1]: Does it work with link tests? I remember some issues here...

Comment (SvyatoslavScherbina) [reply backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt:1]: It should.

Comment (homuroll) [reply backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt:1]: Reports error: `error: could not find 'main' in '<root>' package.`

Comment (homuroll) [reply backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt:1]: Ah, ok, worked after explicitly adding `-tr` flag to a compilation.

Comment (SvyatoslavScherbina) [reply backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt]: This probably just workarounds the original issue

Comment (SvyatoslavScherbina) [reply backend.native/tests/codegen/initializers/sharedVarInInitBlock_main.kt]: https://github.com/JetBrains/kotlin-native/commit/cd6829b25a6162f958b164dc544de7a9826289e7 ?

**Pull request merged**